python/test_coverage/coverage_daemon.py

```diff
@@ -17,26 +17,29 @@
 
 import os
 import imp
+import platform
 
 
 # This is a hack to always refer the main code rather than built zip.
 main_code_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
 daemon = imp.load_source("daemon", "%s/pyspark/daemon.py" % main_code_dir)
 
 if "COVERAGE_PROCESS_START" in os.environ:
-    worker = imp.load_source("worker", "%s/pyspark/worker.py" % main_code_dir)
+    # PyPy with coverage makes the tests flaky, and CPython is enough for coverage report.
+    if "pypy" not in platform.python_implementation().lower():
```

Comment:
nit: The `platform.python_implementation()` returns one of values (‘CPython’, ‘IronPython’, ‘Jython’, ‘PyPy’) rather than a list, so I guess `!=` is engough. or do you have any other concern?

Reply:
I think this is fine. the string comparison is already flaky given that there might be other Python implementation or they might change the string representation. Permissive comparison should be okay.